apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1.Tests/ResourcePrefixHeaderTest.cs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Grpc.Core;
+using System;
+using System.Linq;
+using Xunit;
+
+namespace Google.Cloud.Spanner.V1.Tests
+{
+    public class ResourcePrefixHeaderTest
+    {
+        const string header = "google-cloud-resource-prefix";
+        const string database = "projects/proj/instances/inst/databases/db";
+        const string session = "projects/proj/instances/inst/databases/db/sessions/sess";
+
+        [Fact]
+        public void SetsHeaderOnCreateSession()
+        {
+            var invoker = new FakeCallInvoker();
+            var client = new SpannerClientBuilder { CallInvoker = invoker }.Build();
+            client.CreateSession(new CreateSessionRequest { Database = database });
+            Metadata.Entry entry = Assert.Single(invoker.Metadata, e => e.Key == header);
+            Assert.Equal(database, entry.Value);
+        }
+
+        [Fact]
+        public void SetsHeaderOnBatchCreateSessions()
+        {
+            var invoker = new FakeCallInvoker();
+            var client = new SpannerClientBuilder { CallInvoker = invoker }.Build();
+            client.BatchCreateSessions(new BatchCreateSessionsRequest { Database = database });
+            Metadata.Entry entry = Assert.Single(invoker.Metadata, e => e.Key == header);
+            Assert.Equal(database, entry.Value);
+        }
+
+        [Fact]
+        public void SetsHeaderOnGetSession()
+        {
+            var invoker = new FakeCallInvoker();
+            var client = new SpannerClientBuilder { CallInvoker = invoker }.Build();
+            client.GetSession(new GetSessionRequest { Name = session });
+            Metadata.Entry entry = Assert.Single(invoker.Metadata, e => e.Key == header);
+            Assert.Equal(database, entry.Value);
+        }
+
+        [Fact]
+        public void SetsHeaderOnListSessions()
+        {
+            var invoker = new FakeCallInvoker();
+            var client = new SpannerClientBuilder { CallInvoker = invoker }.Build();
+            client.ListSessions(new ListSessionsRequest { Database = database });
+            // TODO(skuruppu): Figure out why the invoker doesn't have Metadata set.
```

Comment:
I suspect this is because it hasn't actually issued a request yet. List operations are lazy. If you change the above to 

```csharp
client.ListSessions(new ListSessionsRequest { Database = database }).AsRawResponses().First();
```

... then I suspect it'll work. (Or it may throw another exception because your fake doesn't populate the response appropriately... you'll have to see.)

Reply:
Thanks heaps for the suggestion, it works now!